sources/quest_2/chapter_3/step_12/step_12_target.move

```diff
@@ -0,0 +1,51 @@
+use std::signer;
+module robinson::my_shore {
+
+    
+    struct GlobalData has key {
+        nb_tree: u8,
+        has_river: bool,
+        shore_location: address,
+        animal: vector<u8>,
+    }
+
+    struct House has store, drop{
+        no_of_members : u8,
+        house_location : address,
+        house_owner : address
+    }
+
+    fun init_house(s: signer, members: u8, location: address) {
+        assert!(signer::address_of(&s) == @0x42, 0);
+        let _house = House{
+            no_of_members : members,
+            house_location : location,
+            house_owner:signer::address_of(&s),
+        };
+       
+    }
+
+    fun add_trees(data: &mut GlobalData) {
+
+        data.nb_tree = data.nb_tree + 1
+    }
+
+    struct Resources has store, drop {
+    food: u64,
+    log: u64
+}
+
+    fun init_resource() {
+        let _resources = Resources {
+        food: 1,
+        log: 2
+    };
+}
+
+    fun resources_avail( r: &Resources ) : (u64, u64) {
+        let (food, log) = (r.food, r.log);
+        (food, log);
```

Comment:
No `;` if you want to return data here, it's won't compile like that

Reply:
yes missed that, my bad!